catchup/peerSelector.go

```diff
@@ -80,7 +80,13 @@ type peerClass struct {
 	peerClass   network.PeerOption
 }
 
-// the peersRetriever is a subset of the network.GossipNode used to ensure that we can create an instance of the peerSelector
+// PeerClassExported exports peerClass struct
+type PeerClassExported struct {
```

Comment:
Why do you need this? 
I think the createPeerSelector should be refactored to address your needs. You should not be introducing custom peer classes. So you should not be needing this.